js/src/trace.ts

```diff
@@ -0,0 +1,240 @@
+import { _internalGetGlobalState } from "./logger";
+import iso from "./isomorph";
+
+const MAX_FETCH_RETRIES = 8;
+const INITIAL_RETRY_DELAY_MS = 250;
+
+const sleep = (ms: number) =>
```

Comment:
we shuold use existing backoff libraries. but really this logic should be merged with object fetcher